app/services/package-service.js

```diff
@@ -15,5 +17,14 @@ export default ApiBaseService.extend({
 
   createPackage(pkgParams) {
     return this.POST(`/packages`, pkgParams);
+  },
+
+  updatePackage(pkgId, pkgParams) {
+    return this.PUT(`/packages/${pkgId}`, pkgParams);
+  },
+
+  deletePackage(detailType, detailId) {
```

Comment:
@abulsayyad123 name is ambiguous, it's deleting package's detail and the method says as if it is deleting the package.

Reply:
And i think, this method is related to subform detail, should go to `subform-detail-service.js`. Let me know your thoughts.

Reply:
You are correct @abhinav098 about the naming. But my naming convention is according to first parameter. You can read `deletePackage(detailType)` as `deletePackageDetailType`.

Reply:
@abulsayyad123 Imagine someone writes `deletePackage(123)` what would you assume that method does 😄 

I think I agree that the name is weird.

And even if written `deletePackage(detailType)`, it could be understood as `delete the package of the following detail`

However you do make a point that meaning can sometimes come from argument names, but in those scenarios it's better to use named argument. Ruby is really good for that, in JS that would be an object.

example: 

```javascript

// V1

function sendMail(body, author, recipient) { ... }
sendMail('hello', user1, user2); // We don't know who is the author unless we look at the source code of the function

function sendMail(body, context= {}) { ... }

sendMail('hello', { author: user1, recipient: user2 }); // I don't need to look at the source code to understand 
```